### PR TITLE
UI: Log Program scene changes in studio mode

### PR DESCRIPTION
**Pull request by Penwy**
### Description

This adds a log info message whenever the user changes the Program scene in studio mode.

### Motivation and Context

This was requested for troubleshooting purposes.
[Link](https://discord.com/channels/348973006581923840/376042662580715521/1116925630697701377)

### How Has This Been Tested?

Tested on ubuntu 22.04 by being in studio mode and checking that all transition methods (clicking button, double clicking scene name, Tbar slide and quick transitions) correctly trigger the message. 

### Types of changes

 - Tweak (non-breaking change to improve existing functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: UI/window-basic-main-transitions.cpp [hunk 1]
```diff
@@ -746,8 +746,14 @@ void OBSBasic::CreateProgramDisplay()
...
 void OBSBasic::TransitionClicked()
 {
-	if (previewProgramMode)
+	if (previewProgramMode) {
+		OBSScene scene = GetCurrentScene();
+		const char *sceneName =
+			obs_source_get_name(obs_scene_get_source(scene));
+		blog(LOG_INFO, "User switched Program to scene '%s'",
+		     sceneName);
 		TransitionToScene(GetCurrentScene());
```
Comment (gxalpha) [inline UI/window-basic-main-transitions.cpp:755]: You could use "scene" for this as well I think

File: UI/window-basic-main-transitions.cpp [hunk 2]
```diff
@@ -884,6 +890,11 @@ void OBSBasic::TBarReleased()
 		tBar->blockSignals(false);
 		tBarActive = false;
 		EnableTransitionWidgets(true);
+		OBSScene scene = GetCurrentScene();
+		const char *sceneName =
+			obs_source_get_name(obs_scene_get_source(scene));
```
Comment (gxalpha) [inline UI/window-basic-main-transitions.cpp:895]: There is GetCurrentSceneSource to simplify this

File: UI/window-basic-main-transitions.cpp [hunk 3]
```diff
@@ -1451,6 +1462,10 @@ void OBSBasic::ClearQuickTransitions()
 void OBSBasic::QuickTransitionClicked()
 {
 	int id = sender()->property("id").toInt();
+	OBSScene scene = GetCurrentScene();
+	const char *sceneName =
+		obs_source_get_name(obs_scene_get_source(scene));
```
Comment (gxalpha) [inline UI/window-basic-main-transitions.cpp:1467]: There is GetCurrentSceneSource to simplify this

**Comment by norihiro** [comment]
This feature is actually what I want to have.

I realized some cases that do not match the actual transition and logs.

Instead of putting the same code in multiple places, maybe, it would be better to put the log output code inside `TransitionToScene` and to modify `UI/api-interface.cpp` to notify the transition is made through the API (if really need to differentiate the transition was made by a user or through API).

- When clicking `Fade to Black`, log says that the program was switched to the same scene but not saying switching to black.
- There is no log when a user double-clicked in the multiview projector to make program transition. Detailed steps are as below.
  1. Open `Settings`.
  2. Enable `General` / `Studio Mode` / `Transition to scene when double-clicked` and Click `OK`.
  3. Click `View` / `Multiview (Windowed)` in the main menu.
  4. Double-click some scenes to make program transition.

**Comment by Penwy** [comment]
Thank you for taking the time @norihiro.
I had indeed completely forgotten the existence of multiview.
Everything is now under `TransitionToScene`, except the notification of the end of manual Tbar transition, which need to be under `TBarReleased`.
Here's the current proposed behaviour :
- Upon clicking the button, double clicking scene or any non fade to black quick transition : `User switched Program to scene '%s'`
- Upon fading to black : `User faded from scene '%s' to black`
- Upon fading from black : `User faded from black to scene '%s'`
- Upon starting Tbar transition : `User started manual transition to scene '%s'`, then :
  - Upon bringing Tbar to its end : `Manual transition to scene '%s' finished`, or
  - Upon bringing Tbar back to its start : `Manual transition cancelled`

**Comment by PatTheMav** [comment]
Please avoid linking to Discord messages as context - it requires to have an active Discord account _and_ have access to the server, both of which are not a contribution requirement.

As such, either paste the messages as quoted text in the OP or explain the context in your own words, otherwise the context is lost - anyone without access to our servers will not understand which "troubleshooting purposes" are served by the PR.

**Comment by Lain-B** [comment]
Feel free to let @Penwy know if you'd like to see more changes, but for now I'll just merge this as-is. Seems fine to me.

**Pull request merged**